blobstore/wants.go

```diff
@@ -3,52 +3,57 @@ package blobstore
 import (
 	"context"
 	"encoding/json"
+	"fmt"
 	"io"
 	"sync"
 
 	"github.com/cryptix/go/logging"
-	"github.com/davecgh/go-spew/spew"
+	"github.com/go-kit/kit/metrics/prometheus"
 	"github.com/pkg/errors"
-
 	"go.cryptoscope.co/luigi"
 	"go.cryptoscope.co/muxrpc"
+
 	"go.cryptoscope.co/ssb"
 )
 
-func dump(log logging.Interface, v interface{}, from string) {
-	var x interface{}
-	if msg, ok := v.(WantMsg); ok {
-		x = &msg
-	}
-
-	if msg, ok := v.(*WantMsg); ok {
-		m := make(map[string]int64)
-		for _, w := range *msg {
-			m[w.Ref.Ref()] = w.Dist
-		}
-		x = m
-	}
-
-	log.Log("from", from, "blobSpew", spew.Sprint(x))
-}
-
-func NewWantManager(log logging.Interface, bs ssb.BlobStore) ssb.WantManager {
+func NewWantManager(log logging.Interface, bs ssb.BlobStore, opts ...interface{}) ssb.WantManager {
 	wmgr := &wantManager{
 		bs:    bs,
 		wants: make(map[string]int64),
 		info:  log,
 	}
 
+	for i, o := range opts {
+		switch v := o.(type) {
+		case *prometheus.Gauge:
+			wmgr.gauge = v
+		case *prometheus.Counter:
+			wmgr.evtCtr = v
+		default:
+			log.Log("warning", "unhandled option", "i", i, "type", fmt.Sprintf("%T", o))
+		}
+	}
+
+	wmgr.promGaugeSet("proc", 0)
+
 	wmgr.wantSink, wmgr.Broadcast = luigi.NewBroadcast()
 
 	bs.Changes().Register(luigi.FuncSink(func(ctx context.Context, v interface{}, err error) error {
 		wmgr.l.Lock()
 		defer wmgr.l.Unlock()
 
 		n, ok := v.(ssb.BlobStoreNotification)
-		if ok && n.Op == ssb.BlobStoreOpPut {
-			if _, ok := wmgr.wants[n.Ref.Ref()]; ok {
-				delete(wmgr.wants, n.Ref.Ref())
+		if ok {
+			wmgr.promEvent(n.Op.String(), 1)
+			switch n.Op {
+			case ssb.BlobStoreOpPut:
+				if _, ok := wmgr.wants[n.Ref.Ref()]; ok {
+					delete(wmgr.wants, n.Ref.Ref())
+
+					wmgr.promGaugeSet("nwants", len(wmgr.wants))
+				}
+			default:
+				log.Log("evnt", "warn/debug", "msg", "unhandled blobStore change", "notify", n)
```

Comment:
No clear reason.. I just really dislike ignoring other unhanded types somehow.

Reply:
Hmm, I don't see why. We are really only interested in the Put notifications here, and I'm afraid this will add unnecessary log noise. I mean this isn't a general handler for blob store notifications, we only register for the put messages.

Reply:
I see. I guess I just want enums for the Ops.. ^_^